### PR TITLE
cdist: init at 7.0.0

### PR DESCRIPTION
**Pull request by KAction**

### CODE REVIEW

File: pkgs/tools/admin/cdist/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+buildPythonApplication rec {
+  pname = "cdist";
+  version = "7.0.0";
+  outputs = [ "out" "man" "doc" ];
+
+  src = fetchFromGitea {
+    domain = "code.ungleich.ch";
+    owner = "ungleich-public";
+    repo = "cdist";
+    rev = "be6e7fcc08d5c4e98f7ce30b157ee087010cd126";
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:12]: At first glance it wooks like that the source is tagged.

```suggestion
    rev = "${version}";
```

File: pkgs/tools/admin/cdist/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  doCheck = false;
+
+  postInstall = ''
+    mkdir -p $out/share
+    mv docs/dist/man $out/share
+  '';
+
+  meta = with lib; {
+    description = "Minimalistic configuration management system";
+    homepage = "https://sdi.st";
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:43]: I get a 404 with that URL.

```suggestion
    homepage = "https://www.cdi.st/";
```

File: pkgs/tools/admin/cdist/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    changelog = "https://code.ungleich.ch/ungleich-public/cdist/src/branch/master/docs/changelog";
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:44]:
```suggestion
    changelog = "https://code.ungleich.ch/ungleich-public/cdist/src/tag/$version/docs/changelog";
```

File: pkgs/tools/admin/cdist/default.nix [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    # Mostly. There are still couple types that are gpl3-only.
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ kaction ];
+    platforms = lib.platforms.unix;
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:49]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/tools/admin/cdist/default.nix [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+  preConfigure = ''
+    export HOME=/tmp
+  '';
+
+  # Test suite requires either non-chrooted environment or root.
+  #
+  # When "machine_type" explorer figures out that it is running inside
+  # chroot, it assumes that it has enough privileges to escape it.
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:34]: Looks like that we could run `pythonImportsCheck`.

File: pkgs/tools/admin/cdist/default.nix [hunk 6]
```diff
@@ -0,0 +1,51 @@
+{ lib, buildPythonApplication, fetchFromGitea, sphinxHook, sphinx-rtd-theme }:
```
Comment (fabaff) [inline pkgs/tools/admin/cdist/default.nix:1]: One item per line make the diffs easier to read in the future.

```suggestion
{ lib
, buildPythonApplication
, fetchFromGitea
, sphinxHook
, sphinx-rtd-theme
}:
```

**Review by fabaff — COMMENTED** [review]
Upstream says in their documentation that it could also be used as a [library](https://www.cdi.st/manual/latest/cdist-integration.html). I'm not sure how common that use-case is.

**Comment by KAction** [comment]
I think I addressed all comments. Integration library is still experimental, according to upstream:

> WARNING: cdist integration helper functions are not yet stable!

So I'd rather not split binary and library until somebody comes and say that he wants it.

**Comment by KAction** [comment]
@fabaff Can you take another look, please?

**Comment by KAction** [comment]
@happysalada Can you take a look, please?

**Pull request merged**